pintracker/maptracker/maptracker.go

```diff
@@ -271,50 +273,50 @@ func (mpt *MapPinTracker) unpin(c api.Pin) error {
 	}
 
 	mpt.set(c.Cid, api.TrackerStatusUnpinned)
-	mpt.optracker.finish(c.Cid)
+	mpt.optracker.Finish(c.Cid)
 	return nil
 }
 
 // Track tells the MapPinTracker to start managing a Cid,
 // possibly triggering Pin operations on the IPFS daemon.
 func (mpt *MapPinTracker) Track(c api.Pin) error {
 	logger.Debugf("tracking %s", c.Cid)
-
-	if opc, ok := mpt.optracker.get(c.Cid); ok {
-		switch {
-		case opc.op == operationPin:
-			return nil // already ongoing
-		case opc.op == operationUnpin && opc.phase == phaseQueued:
-			mpt.optracker.finish(c.Cid)
-			return nil // cancelled while in queue, all done
-		case opc.op == operationUnpin && opc.phase == phaseInProgress:
-			mpt.optracker.finish(c.Cid)
-			// cancelled while unpinning: continue and trigger pin
-		}
-	}
-
-	if mpt.isRemote(c) {
+	if ptutil.IsRemotePin(c, mpt.peerID) {
 		if mpt.get(c.Cid).Status == api.TrackerStatusPinned {
-			mpt.optracker.trackNewOperation(
+			mpt.optracker.TrackNewOperation(
 				mpt.ctx,
 				c.Cid,
-				operationUnpin,
+				optracker.OperationUnpin,
 			)
 			mpt.unpin(c)
 		}
 		mpt.set(c.Cid, api.TrackerStatusRemote)
 		return nil
 	}
 
-	mpt.optracker.trackNewOperation(mpt.ctx, c.Cid, operationPin)
+	if opc, ok := mpt.optracker.Get(c.Cid); ok {
```

Comment:
Note sure I agree with the re-ordering. i.e. to parallel calls to Track might run NewOperation (both) and trigger two unpins where there should only be one.

Reply:
Hmm yeah not sure what happened there. I fix that up.